airbyte-integrations/connectors/source-asana/source_asana/streams.py

```diff
@@ -177,14 +191,10 @@ def stream_slices(
         cursor_field: List[str] = None,
         stream_state: Mapping[str, Any] = None
     ) -> Iterable[Optional[Mapping[str, Any]]]:
-        tasks_stream = Tasks(authenticator=self.authenticator)
-        tasks_slices = tasks_stream.stream_slices(sync_mode=SyncMode.full_refresh)
-        for task_slice in tasks_slices:
-            for task in tasks_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=task_slice):
-                yield {"task_gid": task["gid"]}
+        yield from self.read_stream(stream_class=Tasks, slice_field="task_gid")
```

Comment:
the `slice_field` param is a little off here -- this method name implies it should just read the stream but the parameter contains some functionality related to slices. Could we rename this method to be something like `read_slices_from_records` or something?

Reply:
Done.